### ISSUE TITLE
Switch to a different provider for weather data

### ISSUE THREAD

**Issue by codeka** [codeka/weather#3]
Apparently weather underground [have shut down access to their API](https://apicommunity.wunderground.com/weatherapi/topics/end-of-service-for-the-weather-underground-api). Weirdly, it's still actually working for me (mostly), but I assume it's just a matter of time before it's gone completely.

It might be a good idea to switch to a different provider of weather data. One that's actually free (so I could honor #2 and actually release an APK).

**Comment by IzzySoft** [comment]
Pointer: On F-Droid we had a discussion on [Do our weather apps need the NonFreeNet AntiFeature?](https://gitlab.com/fdroid/fdroiddata/issues/1439). That discussion mentiones several APIs – those who are NonFree, but also those that are free and thus don't need the AntiFeature.

**Comment by codeka** [comment]
Some candidates:

https://www.rainviewer.com/api.html
- Pro: at least it seems to work
- Pro: nice-looking maps
- Con: [Doesn't have fully global coverage](https://www.rainviewer.com/coverage.html) (but I don't think anything really does)
- Con: Radar images only (would need another API for other parts)
- ? Unclear license

https://openweathermap.org/
- Pro: free, pretty popular (probably won't be shut down soon)
- Con: The weather maps look really ugly

https://www.weatherbit.io/api
- Con: No radar images
- Con: Free tier kinda sucks

https://developer.accuweather.com/apis
- Con: Free tier is only 50 calls/day, very limited
- Con: Imagery is really expensive

https://www.climacell.co/weather-api/
http://api.weather2020.com/

**Comment by IzzySoft** [comment]
Btw: Despite the "Open" in the name, OpenWeatherMap isn't really "open" (on F-Droid, it would raise the `NonFreeNet` AntiFeature). But the same is probably true for at least wheatherbit and accuweather as well. Cannot tell for the others. If that's not a K.O. criterion, you might also wish to take a look at Swiss [Meteotest](https://meteotest.ch/wetter-api), which offers a variety of world-wide weather data via API (CSV, XML, JSON, YAML). I didn't check their conditions, though.

For other services, also [see here](https://nordicapis.com/6-best-free-and-paid-weather-apis/) and [here](https://hackernoon.com/the-top-8-weather-apis-for-2020-7i1f32oj). I didn't dig in deep, though – looks like there's no "real free" (in the sense of FOSS) one in those lists.

**Comment by codeka** [comment]
Yeah, based on the list you linked previously (https://gitlab.com/fdroid/fdroiddata/-/issues/1439) there's basically no weather APIs that would be considered free, with the possible exception of government sources, but then I'd be on the hook for combining them all into something usable (which, from what I can tell, is basically what RainViewer does)... not really feasible for a project that's mostly just for my personal use anyway :)

**Comment by IzzySoft** [comment]
Yepp, that's why I didn't even mention those (and yes, they're the only FOSS ones I know). Could have been something popped up meanwhile I thought. But I'm not really looking, so I will certainly have missed it unless someone put it right on my nose…

**Comment by IzzySoft** [comment]
@codeka as it no longer seems to work (I just tried it today but only got a blank screen), I'll remove the app from my repo for now. Should you find a solution (NonFreeNet is acceptable) and want me to re-list it, just give me a ping – metadata is just archived on my end, so it's easily re-established. :crossed_fingers:
